tests/graph/test_node_link_json.py

```diff
@@ -140,3 +140,54 @@ def test_not_multigraph(self):
         res = rustworkx.node_link_json(graph)
         expected = {"attrs": None, "directed": False, "links": [], "multigraph": False, "nodes": []}
         self.assertEqual(json.loads(res), expected)
+
+    def test_round_trip(self):
+        graph = rustworkx.generators.path_graph(3)
+        res = rustworkx.node_link_json(graph)
+        new = rustworkx.parse_node_link_json_str(res)
+        self.assertIsInstance(new, type(graph))
+        self.assertEqual(new.nodes(), graph.nodes())
+        self.assertEqual(new.weighted_edge_list(), graph.weighted_edge_list())
+        self.assertEqual(new.attrs, graph.attrs)
+
+    def test_round_trip_with_file(self):
```

Comment:
Overall this is super cool! Do you think we should add a round trip (optional) test where we write from NetworkX and them import it in Rustworkx?

Reply:
Sure, that's simple enough to add.

Reply:
This was a good call to add a test for this. The parser is overly strict because it's assuming there was always an `id` field for edges which networkx doesn't always populate in it's node link json. This isn't required in the format, so I'll tweak the parser to make it optional.

Reply:
I fixed this in: https://github.com/Qiskit/rustworkx/pull/1091/commits/8f75081365f7f190d7beabd72251aad2eff42fa5 and added a test.